### PR TITLE
optimize get all tx result interface

### PR DESCRIPTION
**Pull request by lisuxiaoqi**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/rpc/namespaces/eth/stdtx_api.go [hunk 1]
```diff
@@ -2,68 +2,44 @@ package eth
...
-		switch {
-		case len(pendingTxs) <= int(offset):
-			return nil, nil
-		case len(pendingTxs) < int(offset+limit):
-			return pendingTxs[offset:], nil
-		default:
-			return pendingTxs[offset : offset+limit], nil
-		}
-	case rpctypes.LatestBlockNumber:
+	fmt.Println("Try to get from node")
```
Comment (ilovers) [inline app/rpc/namespaces/eth/stdtx_api.go]: should delete this line.

File: app/rpc/namespaces/eth/stdtx_api.go [hunk 2]
```diff
@@ -74,88 +50,34 @@ func (api *PublicEthereumAPI) GetTransactionsWithStdByBlock(blockNrOrHash rpctyp
 	if err != nil {
 		return nil, err
 	}
+	fmt.Println("Try to get Block:", height)
```
Comment (ilovers) [inline app/rpc/namespaces/eth/stdtx_api.go]: should delete this line.

File: app/rpc/namespaces/eth/stdtx_api.go [hunk 3]
```diff
@@ -74,88 +50,34 @@ func (api *PublicEthereumAPI) GetTransactionsWithStdByBlock(blockNrOrHash rpctyp
...
 		}
 
-		if block == nil {
-			// Query block for consensus hash
-			block, err = api.clientCtx.Client.Block(&queryTx.Height)
+		if realTx != nil {
+			txHash := resBlock.Block.Txs[idx].Hash(resBlock.Block.Height)
+			queryTx, err := api.clientCtx.Client.Tx(txHash, false)
 			if err != nil {
+				fmt.Println("failed to get from tx")
```
Comment (ilovers) [inline app/rpc/namespaces/eth/stdtx_api.go]: should delete this line.

File: x/evm/watcher/querier.go [hunk 4]
```diff
@@ -348,38 +351,141 @@ func (q Querier) GetTransactionsByBlockNumber(number, offset, limit uint64) ([]*
...
 	if err != nil {
 		return nil, err
 	}
 
-	var rawTxs []*Transaction
-	// get eth Txs
+	results := make([]*TransactionResult, 0, limit)
+	var ethStart, ethEnd, ethTxLen uint64
+
+	fmt.Println("--Try to get eth tx from db")
```
Comment (ilovers) [inline x/evm/watcher/querier.go]: should delete this line.

File: x/evm/watcher/querier.go [hunk 5]
```diff
@@ -348,38 +351,141 @@ func (q Querier) GetTransactionsByBlockNumber(number, offset, limit uint64) ([]*
...
-	stdTxs, err := q.GetStdTxHashByBlockHash(block.Hash)
+	// enough Tx by Eth
+	ethTxNums := ethEnd - ethStart
+	if ethTxNums == limit {
+		fmt.Println("--Enough Tx")
+		return results, nil
+	}
+	// calc remain std txs
+	remainTxs := limit - ethTxNums
+	fmt.Println("--remainTxs Tx:", remainTxs)
```
Comment (ilovers) [inline x/evm/watcher/querier.go]: should delete this line.

File: x/evm/watcher/querier.go [hunk 6]
```diff
@@ -348,38 +351,141 @@ func (q Querier) GetTransactionsByBlockNumber(number, offset, limit uint64) ([]*
...
+		ethTxLen = uint64(len(txsHash))
+
+		if offset < ethTxLen {
+			ethStart = offset
+			if ethEnd = ethStart + limit; ethEnd > ethTxLen {
+				ethEnd = ethTxLen
+			}
+		}
+
+		fmt.Println("ethStart:", ethStart, "--ethEnd:", ethEnd)
```
Comment (ilovers) [inline x/evm/watcher/querier.go]: should delete this line.

File: x/evm/watcher/querier.go [hunk 7]
```diff
@@ -348,38 +351,141 @@ func (q Querier) GetTransactionsByBlockNumber(number, offset, limit uint64) ([]*
...
+
+	if stdTxsHash != nil && len(stdTxsHash) != 0 {
+		stdTxsLen := uint64(len(stdTxsHash))
+		var stdStart, stdEnd uint64
+		stdStart = offset + ethTxNums - ethTxLen
+		if stdEnd = stdStart + remainTxs; stdEnd > stdTxsLen {
+			stdEnd = stdTxsLen
+		}
+
+		fmt.Println("stdStart:", stdStart, "--stdEnd:", stdEnd)
```
Comment (ilovers) [inline x/evm/watcher/querier.go]: should delete this line.

File: app/rpc/namespaces/eth/stdtx_api.go [hunk 8]
```diff
@@ -2,68 +2,44 @@ package eth
...
 
-	txs, e := api.wrappedBackend.GetTransactionsWithStdByBlockNumber(uint64(blockNum), uint64(offset), uint64(limit))
-	if e == nil && txs != nil {
-		return txs, nil
+	// try to get from watch db
+	fmt.Println("Try to get from db", offset, limit)
+	results, err = api.wrappedBackend.GetTxResultByBlock(api.clientCtx, uint64(blockNum), uint64(offset), uint64(limit))
+	if err == nil && results != nil && len(results) != 0 {
+		return results, nil
 	}
```
Comment (ilovers) [inline app/rpc/namespaces/eth/stdtx_api.go]:
```suggestion
	if err == nil  {
		return results, nil
	}
```

**Review by ilovers — DISMISSED** [review]

File: app/rpc/namespaces/eth/stdtx_api.go [hunk 9]
```diff
@@ -4,158 +4,75 @@ import (
...
-		return nil, err
-	}
-
-	return rpcTx, nil
-}
-
-// GetTransactionsByBlock returns some transactions identified by number or hash.
-func (api *PublicEthereumAPI) GetTransactionsWithStdByBlock(blockNrOrHash rpctypes.BlockNumberOrHash,
-	offset, limit hexutil.Uint) ([]*watcher.Transaction, error) {
+// GetTransactionReceiptsByBlock returns the transaction receipt identified by block hash or number.
```
Comment (xiangjianmeng) [inline app/rpc/namespaces/eth/stdtx_api.go:15]: should be GetTransactionReceiptsByBlock

**Review by xiangjianmeng — DISMISSED** [review]

**Review by xiangjianmeng — APPROVED** [review]

**Pull request merged**